homeassistant/helpers/device_registry.py

```diff
@@ -385,10 +389,11 @@ def _async_update_device(
             ("sw_version", sw_version),
             ("entry_type", entry_type),
             ("via_device_id", via_device_id),
-            ("disabled_by", disabled_by),
         ):
             if value is not UNDEFINED and value != getattr(old, attr_name):
                 changes[attr_name] = value
+        if disabled_by is not UNDEFINED and disabled_by != getattr(old, "disabled_by"):
+            changes["disabled_by"] = disabled_by
```

Comment:
Why do we move this from the for loop?

Reply:
At one point when working on the change, the type of `disabled_by` was different from others handled in the loop, which caused typing problems when it was assigned to `value` that had a different inferred type from earlier loop rounds.

Seems I ended up changing all types handled in the loop to have the same type later, but failed to revert this change, will do so in a push momentarily.